models/base/dim_course_blocks.sql

```diff
@@ -5,7 +5,17 @@ select
     courses.course_run as course_run,
     blocks.location as block_id,
     blocks.block_name as block_name,
-    blocks.display_name_with_location as display_name_with_location
+    blocks.display_name_with_location as display_name_with_location,
+    graded,
+    case
+        when block_id like '%@chapter+block@%'
+        then 'section'
+        when block_id like '%@sequential+block@%'
+        then 'subsection'
+        when block_id like '%@vertical+block@%'
+        then 'unit'
+        else regexpExtract(block_id, '@([^+]+)\+block@', 1)
```

Comment:
Not all block names will end in "block", it's probably best just to take the whole string if it doesn't match one of the above

Reply:
Nevermind, I don't think the cases where that is true are relevant here (at least not yet, it will be things like library block names).